### PR TITLE
Get target information for tabular dataset

### PR DESCRIPTION
**Pull request by sooahleex**
### Summary

- Set `target` as `dictionary` to get input and output column. ex. `{"input": "question", "output":["rating", "sentiment"]}`
- If `target` is `None`, bring all columns
- Set input of target column as media and output of it as annotations
  - For example, if we set target as `{"input": "length(m)", "output": ["breed_category", "pet_category"]}`
    image
    And each item for train and test subset is like below.
    image
    Test file did not have column for `breed_category` and `pet_category`, so test subset did not have any annotations. 

- Set `CategoricalDtype` for column which have dtype as object but that could be used as label. For this, we should define the threshold.

### How to test

- Fix target for unit test and cli test

### Checklist

- [X] I have added unit tests to cover my changes.​
- [X] I have added integration tests to cover my changes.​
- [X] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [X] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: src/datumaro/plugins/data_formats/tabular.py [hunk 1]
```diff
@@ -84,45 +86,56 @@ def _parse(
         items: List[DatasetItem] = []
         categories: TabularCategories = TabularCategories()
 
+        if target is not None:
+            target = eval(",".join(target)) if isinstance(target, list) else target
+            if target["input"] is None or target["output"] is None:
+                raise TypeError('Target should have both "input" and "output"')
```
Comment (wonjuleee) [inline src/datumaro/plugins/data_formats/tabular.py:91]: maybe it is more good for considering `target={}`. In this case, media=TableRow(all_data) and annotations=[]. What do you think?

Comment (sooahleex) [reply src/datumaro/plugins/data_formats/tabular.py:91]: I set `media=TableRow(all_data)` and `annotations = []` when `target = {}`. Fixed in [0c84d9e](https://github.com/openvinotoolkit/datumaro/pull/1471/commits/0c84d9e5673c1832d67092b827ba71507b551152)

Comment (wonjuleee) [reply src/datumaro/plugins/data_formats/tabular.py:91]: Thank you

**Review by wonjuleee — APPROVED** [review]
LGTM

**Pull request merged**